plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -4839,6 +4852,11 @@ private Answer migrateVolume(MigrateVolumeCommand cmd) {
         String path = cmd.getVolumePath();
 
         VmwareHypervisorHost hyperHost = getHyperHost(getServiceContext());
+        VmwareHypervisorHost hyperHostInTargetCluster = null;
+        if (cmd.getHostGuidInTargetCluster() != null) {
+            hyperHostInTargetCluster = VmwareHelper.getHostMOFromHostName(getServiceContext(), cmd.getHostGuidInTargetCluster());
+        }
+        VmwareHypervisorHost targetDSHost = hyperHostInTargetCluster != null ? hyperHostInTargetCluster : hyperHost;
```

Comment:
this merrits a call to a specific method

Reply:
@DaanHoogland which specific method? I didn't get you

Reply:
something like 

```suggestion
        VmwareHypervisorHost targetDSHost = getTargetHostFromTargetCluster(cmd);
```

and

```suggestion
    private VmwareHypervisorHost getTargetHostFromTargetCluster(MigrateVolumeCmd cmd) {
        VmwareHypervisorHost hyperHostInTargetCluster = null;
        if (cmd.getHostGuidInTargetCluster() != null) {
            hyperHostInTargetCluster = VmwareHelper.getHostMOFromHostName(getServiceContext(), cmd.getHostGuidInTargetCluster());
        }
        return hyperHostInTargetCluster != null ? hyperHostInTargetCluster : hyperHost;
    }
```

Reply:
@DaanHoogland both `hyperHostInTargetCluster` and `targetDSHost` are used in the method later multiple times. I used two different variables to prevent using if..else multiple times.